doc/src/Manual_version.rst

```diff
@@ -53,3 +33,48 @@ first page of the :doc:`manual <Manual>`.
 * If you browse the HTML pages included in your downloaded tarball, they
   describe the version you have, which may be older than the online
   version.
+
+Development
+^^^^^^^^^^^
+
+Modifications of the LAMMPS source code (like bug fixes, code
+refactoring, updates to existing features, or addition of new features)
+are organized into pull requests.  Pull requests will be merged into the
+*develop* branch of the git repository after they pass automated testing
+and code review by the LAMMPS developers.
+
+Feature Releases
+^^^^^^^^^^^^^^^^
+
+When a sufficient number of new features and updates have accumulated
+*and* the LAMMPS version on the *develop* branch passes an extended set
+of automated tests, we release it as a *feature release*, which are
+currently made every 4 to 8 weeks.  The *release* branch of the git
+repository is updated with every such *feature release* and a tag in the
+format ``patch_1May2014`` is added.  A summary of the most important
+changes of these releases for the current year are posted on `this
+website page <https://www.lammps.org/bug.html>`_.  More detailed release
+notes are `available on GitHub
+<https://github.com/lammps/lammps/releases/>`_.
+
+Stable Releases
+^^^^^^^^^^^^^^^
+
+About once a year, we release a *stable release* version of LAMMPS.
+This is done after a "stabilization period" where we apply only bug
+fixes and small, non-intrusive changes to the *develop* branch but no
+new features.  At the same time, the code is subjected to more detailed
+and thorough manual testing than the default automated testing.
+After such a *stable release*, both the *release* and the *stable*
+branches are updated and two tags are applied, a ``patch_1May2014`` format
+and a ``stable_1May2014`` format tag.
+
+Stable Release Updates
+^^^^^^^^^^^^^^^^^^^^^^
+
+Between *stable releases*, we collect bug fixes and updates back-ported
+from the *develop* branch in a branch called *maintenance*.  From the
+*maintenance* branch we make occasional *stable update releases* and
+update the *stable* branch accordingly.  The first update to the
+``stable_1May2014`` release would be tagged as
+``stable_1May2014_update1``.  These updates contain no new features.
```

Comment:
This page seems much better altogether. I still think that a table such as I suggested in https://github.com/lammps/lammps/pull/4024 as a sort of summary would be helpful for the lazy readers. I'd even put it in the beginning of the file, as a sort of TL;DR.

Reply:
@doronbehar thanks for your feedback. Please understand that your personal preferences (and frustrations) are rather different from the consensus among the LAMMPS developers. Over the years there must have been many tens of thousands of people that were not having the kind of problems that you had. Thus if we make too drastic changes, the effect may be counterproductive. It is always a struggle between providing complete and detailed information and providing just sufficient information without distracting people too much. For content at the beginning of the manual, the preference is for the latter and I am concerned that the changes I am proposing are already pushing too far. But that is why we have a require review from at least one other LAMMPS developer before changes are applied and I am waiting for that kind of feedback before changes will be included in the distribution.